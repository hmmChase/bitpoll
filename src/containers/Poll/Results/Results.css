.chart {
  margin: 0 0 0 40px;
}

.tally {
  margin-right: 50px;
  position: relative;
  bottom: 10px;
  margin: 0 auto 10px auto;
}

.tally-left {
  margin-right: 47px;
}
