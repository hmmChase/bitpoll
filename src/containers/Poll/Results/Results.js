import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { Bar } from '@vx/shape';
import { Group } from '@vx/group';
import { GradientLightgreenGreen } from '@vx/gradient';
import { AxisBottom } from '@vx/axis';
import { scaleBand, scaleLinear } from '@vx/scale';
import './Results.css';

export const Results = props => {
  const data = [
    {
      option: 'Yes',
      tally: props.option1Tally
    },
    {
      option: 'No',
      tally: props.option2Tally
    }
  ];
  const width = 200;
  const height = 100;
  const margin = { top: 20, bottom: 20, left: 20, right: 20 };
  const x = d => d.option;
  const y = d => +d.tally;
  const xMax = width - margin.left - margin.right;
  const yMax = height - margin.top - margin.bottom;
  const xScale = scaleBand({
    rangeRound: [0, xMax],
    domain: data.map(x),
    padding: 0.4
  });
  const yScale = scaleLinear({
    rangeRound: [yMax, 0],
    domain: [0, Math.max(...data.map(y))]
  });
  const compose = (scale, accessor) => data => scale(accessor(data));
  const xPoint = compose(
    xScale,
    x
  );
  const yPoint = compose(
    yScale,
    y
  );

  return (
    <div>
      <svg className="chart" width={width} height={height}>
        {data.map((d, i) => {
          const barHeight = yMax - yPoint(d);
          return (
            <Group key={`bar-${i}`}>
              <GradientLightgreenGreen id="LightgreenGreen" />
              <AxisBottom scale={xScale} top={yMax} stroke={''} />

              <Bar
                x={xPoint(d)}
                y={yMax - barHeight}
                height={barHeight}
                width={xScale.bandwidth()}
                fill={`url(#LightgreenGreen)`}
              />
            </Group>
          );
        })}
      </svg>
      <div className="tally">
        <span className="tally-left">{props.option1Tally}</span>
        <span>{props.option2Tally}</span>
      </div>
    </div>
  );
};
export const mapStateToProps = state => ({
  option1Tally: state.poll.option1Tally,
  option2Tally: state.poll.option2Tally
});

Results.propTypes = {
  option1Tally: PropTypes.number.isRequired,
  option2Tally: PropTypes.number.isRequired
};

export default connect(mapStateToProps)(Results);
