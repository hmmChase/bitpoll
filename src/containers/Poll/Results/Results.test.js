import React from 'react';
import { shallow } from 'enzyme';
import { Results } from './Results';

describe('Results', () => {
  let results;
  let mockProps;

  beforeEach(() => {
    mockProps = {
      option1Tally: 34,
      option2Tally: 65
    };
    results = shallow(<Results {...mockProps} />);
  });

  it('matches snapshot', () => {
    expect(results).toMatchSnapshot();
  });
});
