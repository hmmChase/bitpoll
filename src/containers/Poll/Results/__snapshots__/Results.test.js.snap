// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Results matches snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    option1Tally={34}
    option2Tally={65}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <svg
          className="chart"
          height={100}
          width={200}
>
          <Group>
                    <Unknown
                              id="LightgreenGreen"
                    />
                    <AxisBottom
                              scale={[Function]}
                              stroke=""
                              top={60}
                    />
                    <Bar
                              fill="url(#LightgreenGreen)"
                              height={31}
                              width={40}
                              x={27}
                              y={29}
                    />
          </Group>
          <Group>
                    <Unknown
                              id="LightgreenGreen"
                    />
                    <AxisBottom
                              scale={[Function]}
                              stroke=""
                              top={60}
                    />
                    <Bar
                              fill="url(#LightgreenGreen)"
                              height={60}
                              width={40}
                              x={93}
                              y={0}
                    />
          </Group>
</svg>,
        <div
          className="tally"
>
          <span
                    className="tally-left"
          >
                    34
          </span>
          <span>
                    65
          </span>
</div>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <Group>
              <Unknown
                            id="LightgreenGreen"
              />
              <AxisBottom
                            scale={[Function]}
                            stroke=""
                            top={60}
              />
              <Bar
                            fill="url(#LightgreenGreen)"
                            height={31}
                            width={40}
                            x={27}
                            y={29}
              />
</Group>,
            <Group>
              <Unknown
                            id="LightgreenGreen"
              />
              <AxisBottom
                            scale={[Function]}
                            stroke=""
                            top={60}
              />
              <Bar
                            fill="url(#LightgreenGreen)"
                            height={60}
                            width={40}
                            x={93}
                            y={0}
              />
</Group>,
          ],
          "className": "chart",
          "height": 100,
          "width": 200,
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "bar-0",
            "nodeType": "function",
            "props": Object {
              "children": Array [
                <Unknown
                  id="LightgreenGreen"
/>,
                <AxisBottom
                  scale={[Function]}
                  stroke=""
                  top={60}
/>,
                <Bar
                  fill="url(#LightgreenGreen)"
                  height={31}
                  width={40}
                  x={27}
                  y={29}
/>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "id": "LightgreenGreen",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "scale": [Function],
                  "stroke": "",
                  "top": 60,
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "fill": "url(#LightgreenGreen)",
                  "height": 31,
                  "width": 40,
                  "x": 27,
                  "y": 29,
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "bar-1",
            "nodeType": "function",
            "props": Object {
              "children": Array [
                <Unknown
                  id="LightgreenGreen"
/>,
                <AxisBottom
                  scale={[Function]}
                  stroke=""
                  top={60}
/>,
                <Bar
                  fill="url(#LightgreenGreen)"
                  height={60}
                  width={40}
                  x={93}
                  y={0}
/>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "id": "LightgreenGreen",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "scale": [Function],
                  "stroke": "",
                  "top": 60,
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "fill": "url(#LightgreenGreen)",
                  "height": 60,
                  "width": 40,
                  "x": 93,
                  "y": 0,
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": [Function],
          },
        ],
        "type": "svg",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <span
              className="tally-left"
>
              34
</span>,
            <span>
              65
</span>,
          ],
          "className": "tally",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": 34,
              "className": "tally-left",
            },
            "ref": null,
            "rendered": 34,
            "type": "span",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": 65,
            },
            "ref": null,
            "rendered": 65,
            "type": "span",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <svg
            className="chart"
            height={100}
            width={200}
>
            <Group>
                        <Unknown
                                    id="LightgreenGreen"
                        />
                        <AxisBottom
                                    scale={[Function]}
                                    stroke=""
                                    top={60}
                        />
                        <Bar
                                    fill="url(#LightgreenGreen)"
                                    height={31}
                                    width={40}
                                    x={27}
                                    y={29}
                        />
            </Group>
            <Group>
                        <Unknown
                                    id="LightgreenGreen"
                        />
                        <AxisBottom
                                    scale={[Function]}
                                    stroke=""
                                    top={60}
                        />
                        <Bar
                                    fill="url(#LightgreenGreen)"
                                    height={60}
                                    width={40}
                                    x={93}
                                    y={0}
                        />
            </Group>
</svg>,
          <div
            className="tally"
>
            <span
                        className="tally-left"
            >
                        34
            </span>
            <span>
                        65
            </span>
</div>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <Group>
                <Unknown
                                id="LightgreenGreen"
                />
                <AxisBottom
                                scale={[Function]}
                                stroke=""
                                top={60}
                />
                <Bar
                                fill="url(#LightgreenGreen)"
                                height={31}
                                width={40}
                                x={27}
                                y={29}
                />
</Group>,
              <Group>
                <Unknown
                                id="LightgreenGreen"
                />
                <AxisBottom
                                scale={[Function]}
                                stroke=""
                                top={60}
                />
                <Bar
                                fill="url(#LightgreenGreen)"
                                height={60}
                                width={40}
                                x={93}
                                y={0}
                />
</Group>,
            ],
            "className": "chart",
            "height": 100,
            "width": 200,
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "bar-0",
              "nodeType": "function",
              "props": Object {
                "children": Array [
                  <Unknown
                    id="LightgreenGreen"
/>,
                  <AxisBottom
                    scale={[Function]}
                    stroke=""
                    top={60}
/>,
                  <Bar
                    fill="url(#LightgreenGreen)"
                    height={31}
                    width={40}
                    x={27}
                    y={29}
/>,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "id": "LightgreenGreen",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "scale": [Function],
                    "stroke": "",
                    "top": 60,
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "fill": "url(#LightgreenGreen)",
                    "height": 31,
                    "width": 40,
                    "x": 27,
                    "y": 29,
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
              ],
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "bar-1",
              "nodeType": "function",
              "props": Object {
                "children": Array [
                  <Unknown
                    id="LightgreenGreen"
/>,
                  <AxisBottom
                    scale={[Function]}
                    stroke=""
                    top={60}
/>,
                  <Bar
                    fill="url(#LightgreenGreen)"
                    height={60}
                    width={40}
                    x={93}
                    y={0}
/>,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "id": "LightgreenGreen",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "scale": [Function],
                    "stroke": "",
                    "top": 60,
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "fill": "url(#LightgreenGreen)",
                    "height": 60,
                    "width": 40,
                    "x": 93,
                    "y": 0,
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
              ],
              "type": [Function],
            },
          ],
          "type": "svg",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <span
                className="tally-left"
>
                34
</span>,
              <span>
                65
</span>,
            ],
            "className": "tally",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": 34,
                "className": "tally-left",
              },
              "ref": null,
              "rendered": 34,
              "type": "span",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": 65,
              },
              "ref": null,
              "rendered": 65,
              "type": "span",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
